### ISSUE TITLE
jinja2 lookup('template', '...') hangs

### ISSUE THREAD

**Issue by bretmartin** [ansible/ansible#11909]
##### Issue Type:
- Bug Report
##### Ansible Version:

```
ansible 2.0.0 (devel e32d887609) last updated 2015/08/10 09:00:05 (GMT -400)
  lib/ansible/modules/core: (detached HEAD dea5f5d15e) last updated 2015/08/10 09:00:05 (GMT -400)
  lib/ansible/modules/extras: (detached HEAD 210ee3febc) last updated 2015/08/07 11:44:45 (GMT -400)
  v1/ansible/modules/core: (detached HEAD f8d8af17cd) last updated 2015/08/06 14:46:13 (GMT -400)
  v1/ansible/modules/extras: (detached HEAD 495ad450e5) last updated 2015/08/06 14:46:16 (GMT -400)
  config file =
  configured module search path = None
```
##### Ansible Configuration:

Stock `ansible.cfg`
##### Environment:

N/A but running on Mac OS X
##### Summary:

When a template contains a `lookup('template', '...')`, playbook execution hangs.
##### Steps To Reproduce:

`roles/test/tasks/main.yml`

```

---

- set_fact:
    fact: '{{ lookup("template",
                     playbook_dir + "/roles/test/templates/test.j2") }}'
```

`roles/test/templates/test.j2`

```
{% include "test2.j2" %}
```

`roles/test/templates/test2.j2`

```
hello there
```
##### Expected Results:

Complete playbook execution with fact set, like this run against tag `v1.9.2-1`:

```
PLAY [localhost] **************************************************************

GATHERING FACTS ***************************************************************
ok: [localhost]

TASK: [test | set_fact ] ******************************************************
ok: [localhost]

PLAY RECAP ********************************************************************
localhost                  : ok=2    changed=0    unreachable=0    failed=0
```
##### Actual Results:

```
PLAY ***************************************************************************

TASK [setup] *******************************************************************
ok: [localhost]

TASK [test : set_fact fact={{ lookup("template", playbook_dir + "/roles/test/templates/test.j2") }}] ***
[ indefinite hang ]
```

**Comment by bretmartin** [comment]
I mischaracterized this bug -- it is actually the lookup() that appears to be causing the hang.

**Comment by bretmartin** [comment]
My change in #11929 seems to correct this, although I understand from @bcoca that may not be a proper fix overall.

**Comment by bretmartin** [comment]
Confirmed to still be an issue as of d9833f227f. Definitely related to #11929 -- the template lookup plugin is trying to use the first character of as the file name. With `-vvvv`:

```
TASK [test-11909 : set_fact fact={{ lookup("template", playbook_dir + "/roles/test/templates/test.j2") }}] ***
File lookup using / as file
```

**Comment by bcoca** [comment]
fixed via 2c34a5bdbd0685f449bbb076401b050276094776

**Comment by bretmartin** [comment]
@bcoca Fix confirmed here. Thank you!
